### PR TITLE
Internationalisation support

### PR DESCRIPTION
**Pull request by stegru**
This adds support for i18n.

We'll have a slow-moving background task of transforming the additional pages so they're localisable, but once this PR is in everything new should be localisable. Meaning, all text should be placed into `src/locales/en.json`.

To get the ball rolling, I've already translated ResetPassword.vue. 

The README.md, pasted below, has more info:

# Internationalisation

## Files

* [src/i18n/i18n.js](i18n.js) - Detects the locale, and loads the messages. (Invoked from [App.vue](../App.vue))
* [src/locales/*.json](../locales) - The locale message files.
* [src/config/locales.js](../config/locales.js) - Supported locales.

## Guide

* [vue-i18n](https://kazupon.github.io/vue-i18n) is used.
* Strings in the messages files are grouped by the file they are referenced in.
* Keys should describe the text, rather than duplicate it.
* Prefer to not re-use the strings for different elements, even if the text is the same. Instead, [Linked messages](https://kazupon.github.io/vue-i18n/guide/messages.html#linked-locale-messages)
  can be used, so the translator can override if required.
* For simple text, the `v-t` directive is preferred, because `$t()` gets evaluated on every re-render.

### Tooling

* IntelliJ IDEA: [i18nPlugin](https://github.com/nyavro/i18nPlugin)
* VS Code: [i18n-ally](https://github.com/lokalise/i18n-ally)

Regex to replace simple `$t` methods, where it is the only content of an element, with the `v-t` directive:

* / `(?<TAG><(\S+)[^>]*)>\{\{ *\$t\((?<KEY>'[^']*')\) }}(</\2>)` / `${TAG} v-t="${KEY}" />` /
* `<b class="a">{{ $t('hello-text') }}</b>` => `<b class="a" v-t="'hello-text'" />`

## Examples

Messages file

```json5
{
    "example-page": {
        // plain text
        "heading": "I18N example",
        // Text for an attribute
        "first-name_placeholder": "Your name",
        // Interpolation
        "greeting": "Hello, {name}",
        // Plurals
        "characters": "nothing entered | only one character | {n} characters",
        // Link message
        "product-name": "Morphic",
        "welcome-message": "Welcome to @:example-page.product-name"
    }
}
```

Input HTML

```html
<!-- plain text -->
<h1 v-t="'example-page.heading'" />

<!-- inside an attribute -->
<input :placeholder="$t('example-page.first-name_placeholder')" v-model="firstName" />

<!-- Interpolation -->
<p>{{ $t('example-page.greeting', { name: firstName }) }}</p>

<!-- Plurals -->
<p>{{ $tc('example-page.characters', firstName.length) }}</p>

<!-- Linked message -->
<p v-t="'example-page.welcome-message'" />
```

Result

```html
<h1>I18N example</h1>

<input placeholder="Your name" />

<p>Hello, abc</p>

<p>3 characters</p>

<p>Welcome to Morphic</p>
```

### CODE REVIEW

File: src/i18n/i18n.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * @return {Promise<String>} Resolves when loaded.
+ */
+export async function loadLocaleMessagesAsync(overrideLocale) {
+    let locale = overrideLocale || i18n.locale;
+
+    if (!loadedLanguages.includes(locale)) {
+
+        try {
+            const messages = await import(/* webpackChunkName: "locale-[request]" */ "@/locales/" + locale + ".json");
+
```
Comment (sulaiman-rtf) [inline src/i18n/i18n.js:22]: This console is it intentional

File: src/i18n/i18n.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            console.error("set", locale, messages);
+            i18n.setLocaleMessage(locale, messages);
+            loadedLanguages.push(locale);
+        } catch (err) {
+            console.error(`Unable to load locale '${locale}':`, err);
```
Comment (sulaiman-rtf) [inline src/i18n/i18n.js:27]: console

**Review by sulaiman-rtf — APPROVED** [review]
Looks good to me i will merge if you confirm that the console is intentional

File: src/i18n/i18n.js (hunk 1 above)
Comment (stegru) [reply src/i18n/i18n.js:22]: no - removed

File: src/i18n/i18n.js (hunk 2 above)
Comment (stegru) [reply src/i18n/i18n.js:26]: the error one is ok

**Pull request merged**